tools/build_everything.py

```diff
@@ -146,8 +146,12 @@
         base_source_paths = options.source_dir
         if not base_source_paths:
             base_source_paths = ['.']
-        
-        all_tests = find_tests(base_source_paths[0])
+       
+        all_tests = {}
+        for platform in platforms:
+            for toolchain in toolchains:
+                platform_toolchain_tests = find_tests(base_source_paths[0], platform, toolchain)
+                all_tests.update(platform_toolchain_tests)
```

Comment:
Because tests now change between targets and toolchains, this really needs to be moved above the `build_tests` call inside the main build loop. Otherwise this will try to compile tests that aren't valid for certain combinations of targets and toolchains.